docs/src/topics/sparse_matrix.md

```diff
@@ -0,0 +1,191 @@
+# Sparsity pattern and sparse matrix
+
+An important property of the finite element method is that it results in *sparse matrices*
+for the linear systems to be solved. On this page the topic of sparsity and sparse matrices
+are discussed.
+
+```@contents
+Pages = ["sparse_matrix.md"]
+Depth = 2:2
+```
+
+## Sparsity pattern
+
+The sparse structure of the linear system depends on many factors such as e.g. the weak
+form, the discretization, and the choice of interpolation(s). In the end it boils down to
+how the degrees of freedom (DoFs) *couple* with each other. The most common reason that two
+DoFs couple is because they simply belong to the same element. Note, however, that this is
+not guaranteed to result in a coupling since it depends on the specific weak form that is
+being discretized, see [XXX](@ref). Boundary conditions and constraints can also result in
+additional DoF couplings.
+
+If DoFs `i` and `j` couple, then the computed value in the eventual matrix will be
+*nonzero*. In this case the entry `(i, j)` should be included in the sparsity pattern.
```

Comment:
Is it worth saying something like "structurally non-zero"? Like, it could happen for various reasons that it becomes zero anyway.